app/scripts/lib/state-utils.ts

```diff
@@ -1,34 +1,45 @@
 import { SnapControllerState } from '@metamask/snaps-controllers';
-import { Snap } from '@metamask/snaps-utils';
+import { isSnapId, Snap } from '@metamask/snaps-utils';
+import { BackgroundStateProxy } from '../../../shared/types/metamask';
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-type FlattenedUIState = Record<string, any>;
+const REMOVE_KEYS = ['snapStates', 'unencryptedSnapStates', 'vault'] as const;
 
-const REMOVE_KEYS = ['snapStates', 'unencryptedSnapStates', 'vault'];
-
-export function sanitizeUIState(state: FlattenedUIState): FlattenedUIState {
+export function sanitizeUIState(
+  state: BackgroundStateProxy,
+): BackgroundStateProxy {
   const newState = { ...state };
 
   for (const key of REMOVE_KEYS) {
-    delete newState[key];
+    if (key === 'vault') {
+      if (newState.KeyringController && key in newState.KeyringController) {
+        delete newState.KeyringController[key];
+      }
+    } else if (newState.SnapController && key in newState.SnapController) {
```

Comment:
The purpose of the `REMOVE_KEYS` constant was to keep this logic generic and not hardcoded.

Could we replace it with `REMOVE_PATHS` instead so we can include the controller keys and keep this logic generic?

Reply:
Sounds sensible. Applied here: https://github.com/MetaMask/metamask-extension/pull/28723/commits/2e251d5d88f6921bdfe861364e2caaaf4830a383